openlibrary/catalog/add_book/tests/test_add_book.py

```diff
@@ -1179,3 +1184,50 @@ def test_add_identifiers_to_edition(mock_site) -> None:
     e = mock_site.get(reply['edition']['key'])
     assert e.works[0]['key'] == '/works/OL19W'
     assert e.identifiers._data == {'goodreads': ['1234'], 'librarything': ['5678']}
+
+
+@pytest.mark.parametrize(
+    'name,rec,error,expected',
+    [
+        (
+            "Books that are too old can't be imported",
+            {'title': 'a book', 'source_records': ['ia:ocaid'], 'publish_date': '1499'},
+            PublicationYearTooOld,
+            None,
+        ),
+        (
+            "But 1500 CE+ can be imported",
+            {'title': 'a book', 'source_records': ['ia:ocaid'], 'publish_date': '1500'},
+            None,
+            None,
+        ),
+        (
+            "But trying to import a book from a future year raises an error",
+            {'title': 'a book', 'source_records': ['ia:ocaid'], 'publish_date': '3000'},
+            PublishedInFutureYear,
+            None,
+        ),
+        (
+            "Independently published books can't be imported",
+            {
+                'title': 'a book',
+                'source_records': ['ia:ocaid'],
+                'publishers': ['Independently Published'],
+            },
+            IndependentlyPublished,
+            None,
+        ),
+        (
+            "Can't import sources that require an ISBN",
+            {'title': 'a book', 'source_records': ['amazon:amazon_id'], 'isbn_10': []},
+            SourceNeedsISBN,
+            None,
+        ),
+    ],
+)
+def test_validate_record(name, rec, error, expected) -> None:
+    if error:
+        with pytest.raises(error):
+            validate_record(rec)
+    else:
+        assert validate_record(rec) is expected, f"Assertion failed for test: {name}"  # type: ignore [func-returns-value]
```

Comment:
Good point! I made this change.

Reply:
I have somewhat un-made this change, as I changed the test to remove the `expected` value, since nothing is expected currently. That said, if you look at the current form and think the `expected` value should return, I will happily resurrect it and the `==` comparison.